fog/view/server/src/db_fetcher.rs

```diff
@@ -200,13 +202,14 @@ where
         num_queued_records_limiter: Arc<(Mutex<usize>, Condvar)>,
         readiness_indicator: ReadinessIndicator,
         sharding_strategy: SS,
+        block_tracker: Arc<Mutex<BlockTracker>>,
         logger: Logger,
     ) {
         let thread = Self {
             db,
             stop_requested,
             shared_state,
-            block_tracker: BlockTracker::new(logger.clone()),
```

Comment:
I am skeptical of the idea to fuse these two block trackers, there are two block trackers because they are tracking different things, and merging them seems likely to lead to bugs.

One block tracker is tracking which blocks we have fetched from the database, and which blocks we still need to fetch.

The other block tracker is tracking which blocks we have actually loaded into the enclave.

If we lose this distinction, bad things happen.
If a block is fetched from DB and not yet loaded into enclave, and there is only one block tracker, then either this block tracker thinks this block is processed, or it isn't.

* If the block tracker doesn't think it is processed, then the DB fetcher will decide that we need to fetch it again, so now we can fetch blocks over and over again. This is bad.
* If the block tracker thinks it is processed, even though it's not in the enclave yet, then when we have to report the "highest_processed_block_count" to the user, the number we report will be wrong, and the user can computer an incorrect balance, because we will have broken the fog view protocol.

So I don't see right now there is any way that merging the block trackers can work. But, I'm willing to be wrong.

Reply:
Just to double check this is comment is relating to an earlier version that fully removed the `block_tracker` from `DbFetcherThread`?  Looking at the current change it seems that there are still 2 trackers.

Reply:
Yes, this is an outdated comment. Graphite chose to link the new changes to this outdated PR.